### PR TITLE
Add tests for Redux selectors

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Add tests for Redux selectors.

JIRA: [EDUCATOR-5779](https://openedx.atlassian.net/browse/EDUCATOR-5779)

**What changed?**

- Add tests for Redux selectors.
- Removed unused selector, `typeOfSelectedAssignment`

For comparison, here are before/after test coverage results as of 87f5c8c

File | % Stmts | % Branch | % Funcs | % Lines 
 ---| ---| ---| ---| ---
 src/data/selectors (before) | 20.63 | 3.08 | 9.43 | 23.15 
 src/data/selectors (after) | 98.33 | 96.61 | 96.08 | 100 

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [x] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

- Refactor only: Follow smoke test plan to verify existing behavior continues to work.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: src/data/selectors/filters.test.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    grade_description: '(0.00/0.00)',
+    module_id: 'block-v1:edX+type@sequential+block@bcdef',
+    label: 'LAB 01',
+    category: 'Labs',
+  },
+];
+
+const gradesData = {
+  results: [
+    {
```
Comment (muselesscreator) [inline src/data/selectors/filters.test.js:59]: v minor nit: could be oneline.  only needs splitout when you are adding multiple keys.
```
const gradesData = { results: [{ section_breakdown: sectionBreakdowns }] };
```

File: src/data/selectors/assignmentTypes.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import selectors from './assignmentTypes';
+
+describe('areGradesFrozen', () => {
+  it('returns whether grades are frozen', () => {
+    let areGradesFrozen = selectors.areGradesFrozen({ assignmentTypes: { areGradesFrozen: true } });
```
Comment (muselesscreator) [inline src/data/selectors/assignmentTypes.test.js:5]: I would pass an arbitrary value when testing a pass-through selector, rather than multiple value tests.

File: src/data/selectors/grades.test.js [hunk 3]
```diff
@@ -15,6 +15,199 @@ const genericHistoryRow = {
...
+  it('returns error string when bulkManagement encounters an error', () => {
+    const result = selectors.bulkImportError({ grades: { bulkManagement: { errorMessages: ['It\'s over 9000', 'Neutrino-triggered bit flips'] } } });
+    expect(result).toEqual('Errors while processing: It\'s over 9000, Neutrino-triggered bit flips');
+  });
+});
+
+describe('grade formatters', () => {
+  const selectedAssignment = { assignmentId: 'block-v1:edX+type@sequential+block@abcde' };
+
+  describe('formatMinAssignmentGrade', () => {
```
Comment (muselesscreator) [inline src/data/selectors/grades.test.js:71]: should note in the test that this is testing all 4 selectors

File: src/data/selectors/special.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import selectors from './special';
+
+describe('hasSpecialBulkManagementAccess', () => {
+  // Copy & restore process for testing purposes
+  const OLD_ENV = process.env;
+  const allowedCourses = 'edX/DemoX/2021T1,edX/DemoX/2021T2';
+
+  beforeEach(() => {
+    jest.resetModules();
```
Comment (muselesscreator) [inline src/data/selectors/special.test.js:9]: what is the purpose of this reset?  I've only used this when programatically importing.

**Review by muselesscreator — COMMENTED** [review]
These look great.  A couple of formatting nits and one big note: too many magic strings/values.  Where possible its much cleaner to have the tests checking against variables than locally/manually-typed strings/value.

**Comment by nsprenkle** [comment]
> These look great. A couple of formatting nits and one big note: too many magic strings/values. Where possible its much cleaner to have the tests checking against variables than locally/manually-typed strings/value.

@muselesscreator, the magic strings should be mostly removed and ready for re-review.

**Review by muselesscreator — APPROVED** [review]
:-) These look great

**Review by leangseu-edx — APPROVED** [review]
